secretmap.go

```diff
@@ -83,13 +83,21 @@ func (m *SecretMap) Put(key string, value Secret, updated time.Time) {
 func (m *SecretMap) Delete(key string) {
```

Comment:
I believe we need to take two codepaths into account. The one which gets hit when listing secrets and the one which gets hit when requesting a specific secret. Before doing all that, I would probably refactor secretmap.go a little. Right now, the fact that a "secret exists but we don't have its value" vs "a secret exists and we have a value" is managed in multiple places.